src/components/cc-product-list/products-controller.test.js

```diff
@@ -0,0 +1,1483 @@
+/* eslint-env node, mocha */
+/**
+ * @typedef {import('./cc-product-list.types.js').Product} Product
+ * @typedef {import('hanbi').Stub<() => void>} Stub
+*/
+
+import { expect } from '@bundled-es-modules/chai';
+import * as hanbi from 'hanbi';
+import { ProductsController } from './products-controller.js';
+
+/**
+ * @param {Partial<Product>} product
+ * @returns {Product}
+ */
+function generateProduct ({
+  name = '',
+  description = '',
+  keywords = [],
+}) {
```

Comment:
question: is this ok to have a `@param` named `product` but it's directly destructured without being named?

Reply:
yes you can, TS will understand it. However, `Partial` is important here as we don't provide everything from the `Product` type.